src/users/usersReducer.js

```diff
@@ -32,10 +33,59 @@ export default (
     case EVENT_USER_REMOVE:
       return state; // TODO
 
-    case EVENT_USER_UPDATE:
-      return state.map(user =>
-        user.user_id === action.userId ? { ...user, ...action.person } : user,
-      );
+    case EVENT: {
+      const { event } = action;
+      switch (event.type) {
+        case EventTypes.realm_user: {
+          switch (event.op) {
+            case 'update': {
+              return state.map(user => {
+                const { person } = event;
+                if (user.user_id !== person.user_id) {
+                  return user;
+                }
+                if (person.custom_profile_field) {
+                  return {
+                    ...user,
+                    profile_data: (() => {
+                      if (person.custom_profile_field.value !== null) {
+                        return {
+                          ...(user.profile_data: $PropertyType<User, 'profile_data'>),
+                          [person.custom_profile_field.id]: ({
+                            value: person.custom_profile_field.value,
+                            rendered_value: person.custom_profile_field.rendered_value,
+
+                            // This assertion is cumbersome. But it fills a
+                            // gap in Flow's coverage…that apparently Flow
+                            // doesn't announce by marking anything with
+                            // `any`. Remove when doing so doesn't stop Flow
+                            // from catching something wrong on
+                            // `value` or `rendered_value`.
+                          }: $Values<$NonMaybeType<$PropertyType<User, 'profile_data'>>>),
```

Comment:
nit: I like to mark situations like this with something like `FlowIssue` (no `$`) in the comment. That helps in finding it, either for a broad sweep for issues we've run into in Flow (perhaps to consider dropping the workarounds after a Flow upgrade occasionally?)… or for finding a specific issue one is thinking of but doesn't remember precisely where we encountered it.